protocol/x/vault/keeper/orders.go

```diff
@@ -10,3 +33,130 @@ import (
+	// Get clob pair, perpetual, market parameter, and market price that correspond to this vault.
+	clobPair, exists := k.clobKeeper.GetClobPair(ctx, clobtypes.ClobPairId(vaultId.Number))
+	if !exists {
+		return orders
+	}
+	perpId := clobPair.Metadata.(*clobtypes.ClobPair_PerpetualClobMetadata).PerpetualClobMetadata.PerpetualId
+	perpetual, err := k.perpetualsKeeper.GetPerpetual(ctx, perpId)
+	if err != nil {
+		return orders
+	}
+	marketParam, exists := k.pricesKeeper.GetMarketParam(ctx, perpetual.Params.MarketId)
+	if !exists {
+		return orders
+	}
+	marketPrice, err := k.pricesKeeper.GetMarketPrice(ctx, perpetual.Params.MarketId)
+	if err != nil {
+		return orders
+	}
+
+	// Get vault (subaccount 0 of corresponding module account).
+	vault := satypes.SubaccountId{
+		Owner:  vaultId.ToModuleAccountAddress(),
+		Number: 0,
+	}
+	// Calculate spread.
+	spreadPpm := lib.Max(
+		MIN_BASE_SPREAD_PPM,
+		BASE_SPREAD_MIN_PRICE_CHANGE_PREMIUM_PPM+marketParam.MinPriceChangePpm,
+	)
+	// Get market price in subticks.
+	subticks := clobtypes.PriceToSubticks(
+		marketPrice,
+		clobPair,
+		perpetual.Params.AtomicResolution,
+		lib.QuoteCurrencyAtomicResolution,
+	)
+	// Get order expiration time.
+	goodTilBlockTime := &clobtypes.Order_GoodTilBlockTime{
+		GoodTilBlockTime: uint32(ctx.BlockTime().Unix()) + ORDER_EXPIRATION_SECONDS,
+	}
+	// Construct one ask and one bid for each layer.
+	askSubticks := new(big.Rat).Set(subticks)
+	bidSubticks := new(big.Rat).Set(subticks)
+	for layer := uint8(1); layer <= NUM_LAYERS; layer++ {
+		// Calculate ask and bid subticks for this layer.
+		askSubticks = lib.BigRatMulPpm(askSubticks, lib.OneMillion+spreadPpm)
+		bidSubticks = lib.BigRatMulPpm(bidSubticks, lib.OneMillion-spreadPpm)
+
+		// Construct ask at this layer.
+		ask := clobtypes.Order{
+			OrderId: clobtypes.OrderId{
+				SubaccountId: vault,
+				ClientId:     k.GetVaultClobOrderClientId(ctx, clobtypes.Order_SIDE_SELL, uint8(layer)),
+				OrderFlags:   clobtypes.OrderIdFlags_LongTerm,
+				ClobPairId:   clobPair.Id,
+			},
+			Side:     clobtypes.Order_SIDE_SELL,
+			Quantums: clobPair.StepBaseQuantums, // TODO (TRA-144): Implement order size
+			Subticks: lib.BigRatRoundToNearestMultiple(
+				askSubticks,
+				clobPair.SubticksPerTick,
+				true, // round up for asks
+			),
+			GoodTilOneof: goodTilBlockTime,
+		}
+
+		// Construct bid at this layer.
+		bid := clobtypes.Order{
+			OrderId: clobtypes.OrderId{
+				SubaccountId: vault,
+				ClientId:     k.GetVaultClobOrderClientId(ctx, clobtypes.Order_SIDE_BUY, uint8(layer)),
+				OrderFlags:   clobtypes.OrderIdFlags_LongTerm,
+				ClobPairId:   clobPair.Id,
+			},
+			Side:     clobtypes.Order_SIDE_BUY,
+			Quantums: clobPair.StepBaseQuantums, // TODO (TRA-144): Implement order size
+			Subticks: lib.BigRatRoundToNearestMultiple(
+				bidSubticks,
+				clobPair.SubticksPerTick,
+				false, // round down for bids
+			),
+			GoodTilOneof: goodTilBlockTime,
+		}
+
+		orders = append(orders, ask, bid)
```

Comment:
nit: Non-blocking, given you already know the number of layers, the slice could have been allocated with length 2*layers and this could be:
```
orders[2*i] = ask
orders[2*i+1] = bid
```